docs/java/how-to-develop-a-workflow-definition-in-java.md

```diff
@@ -146,12 +231,154 @@ Workflow methods annotated with `@QueryMethod` and `@SignalMethod` can be invoke
-Note that, during replay, successfully executed Activities are not re-executed because their results are already recorded
-in the Event History of the Workflow Execution.
+Use `DynamicWorkflow` to implement Workflow Types dynamically. When you register a Workflow implementation type that extends `DynamicWorkflow`, it can be used to implement any Workflow type that is not implicitly registered with the Worker.
+
+The main use case for `DynamicWorkflow` is an implementation of custom Domain Specific Languages (DSLs). A single implementation can implement a Workflow Type which by definition is dynamically loaded from some external source.
+You can also use `DynamicWorkflow` when you need a default Workflow that can handle all Workflow Types that are not registered with a Worker.
+
+The Dynamic Workflow interface is implemented with the `execute` method. This method takes in `EncodedValues` that are inputs to the Workflow Execution. These inputs can be specified by the Client when invoking the Workflow Execution.
+
+```Java
+public class MyDynamicWorkflow implements DynamicWorkflow {
+   @Override
+    public Object execute(EncodedValues args) {
+    }
+}
+```
+
+The `DynamicWorkflow` must be registered with a Worker.
+The following rules apply when registering a Dynamic Workflow with a Worker:
+
+- Only one Workflow implementation that extends `DynamicWorkflow` can be registered with a Worker.
+- You can register multiple type-specific Workflow implementations alongside a single `DynamicWorkflow` implementation.
+- Implement the `execute` method for Dynamic Workflow implementations. Do not specify a `@WorkflowMethod` when using Dynamic Workflows.
+- Implement Signals and Queries dynamically.
+
+  ```Java
+        Workflow.registerListener(
+          (DynamicSignalHandler)
+              (signalName, encodedArgs) -> name = encodedArgs.get(0, String.class));
+  ```
+
+- Since `DynamicWorkflow` can be invoked for different Workflow Types, to check what type is running when your Dynamic Workflow `execute` method runs, use:
+
+  ```Java
+  String type = Workflow.getInfo().getWorkflowType();
+  ```
+
+- All standard `WorkflowOptions` apply to Dynamic Workflows.
+- All the determinism rules apply to Workflows that implement this interface.
+
+The following example shows a Dynamic Workflow Implementation.
+
+```Java
+// Dynamic Workflow Implementation
+public static class DynamicGreetingWorkflowImpl implements DynamicWorkflow {
+  private String name;
+
+  @Override
+  public Object execute(EncodedValues args) {
+    String greeting = args.get(0, String.class);
+    String type = Workflow.getInfo().getWorkflowType();
+
+    // Register dynamic Signal handler
+    Workflow.registerListener(
+        (DynamicSignalHandler)
+            (signalName, encodedArgs) -> name = encodedArgs.get(0, String.class));
+```
+
+The following example shows how to register the Dynamic Workflow implementation with a Worker.
```

Comment:
Missing maybe " and client code how to start a workflow execution".

Reply:
Updated!